api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -67,51 +69,54 @@ type OpenStackDataPlaneRoleSpec struct {
-// +kubebuilder:resource:shortName=osdprole;osdproles
+//+operator-sdk:csv:customresourcedefinitions:displayName="OpenStack Data Plane NodeSet"
+// +kubebuilder:resource:shortName=osdpns;osdpnodeset;osdpnodesets
 //+kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[0].status",description="Status"
 //+kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[0].message",description="Message"
 
-// OpenStackDataPlaneRole is the Schema for the openstackdataplaneroles API
-type OpenStackDataPlaneRole struct {
+// OpenStackDataPlaneNodeSet is the Schema for the openstackdataplanenodesets API
+type OpenStackDataPlaneNodeSet struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   OpenStackDataPlaneRoleSpec `json:"spec,omitempty"`
-	Status OpenStackDataPlaneStatus   `json:"status,omitempty"`
+	Spec   OpenStackDataPlaneNodeSetSpec `json:"spec,omitempty"`
+	Status OpenStackDataPlaneStatus      `json:"status,omitempty"`
 }
 
 //+kubebuilder:object:root=true
 
-// OpenStackDataPlaneRoleList contains a list of OpenStackDataPlaneRole
-type OpenStackDataPlaneRoleList struct {
+// OpenStackDataPlaneNodeSetList contains a list of OpenStackDataPlaneNodeSets
+type OpenStackDataPlaneNodeSetList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []OpenStackDataPlaneRole `json:"items"`
+	Items           []OpenStackDataPlaneNodeSet `json:"items"`
 }
 
 func init() {
-	SchemeBuilder.Register(&OpenStackDataPlaneRole{}, &OpenStackDataPlaneRoleList{})
+	SchemeBuilder.Register(&OpenStackDataPlaneNodeSet{}, &OpenStackDataPlaneNodeSetList{})
 }
 
 // IsReady - returns true if the DataPlane is ready
-func (instance OpenStackDataPlaneRole) IsReady() bool {
-	return instance.Status.Conditions.IsTrue(condition.ReadyCondition)
+func (instance OpenStackDataPlaneNodeSet) IsReady() bool {
+	return instance.Status.Conditions.IsTrue(condition.DeploymentReadyCondition)
 }
 
 // InitConditions - Initializes Status Conditons
-func (instance *OpenStackDataPlaneRole) InitConditions() {
+func (instance *OpenStackDataPlaneNodeSet) InitConditions() {
 	instance.Status.Conditions = condition.Conditions{}
 
 	cl := condition.CreateList(
 		condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(condition.InputReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(SetupReadyCondition, condition.InitReason, condition.InitReason),
+		condition.UnknownCondition(NodeSetBareMetalProvisionReadyCondition, condition.InitReason, condition.InitReason),
+		condition.UnknownCondition(NodeSetIPReservationReadyCondition, condition.InitReason, condition.InitReason),
+		condition.UnknownCondition(NodeSetDNSDataReadyCondition, condition.InitReason, condition.InitReason),
```

Comment:
This partially reverts this commit 28cbe4ec6e88439d46865de44c444dad246ec5fe. Why?

Reply:
Reverting my own hard work. I'll fix this